### PR TITLE
mcuboot-related STM32 fix

### PR DESCRIPTION
**Pull request by mbolivar**
This series helps Zephyr be chain-loaded by mcuboot on STM32 again. (This is currently broken on at least 96b_carbon for multiple reasons, some of which need to be addressed in other pull requests).

### CODE REVIEW

**Comment by mbolivar** [comment]
An implementation which was trying to be more careful about power management might track whether HSI was turned on to serve as SYSCLK source while the PLL was configured, and turn it back off again after switching to the PLL if the PLL's source was HSE or some other non-HSI clock.

I thought about doing that, but it's not clear to me what Zephyr's PM plans are, and I thought I'd keep this simple for now. Please let me know if that was the wrong call.

**Review by erwango — APPROVED** [review]

File: drivers/clock_control/stm32_ll_clock.c [hunk 1]
```diff
@@ -190,7 +190,14 @@ static int stm32_clock_control_init(struct device *dev)
 	/* configure PLL input settings */
 	config_pll_init(&s_PLLInitStruct);
 
-	/* Disable PLL before configuration */
+	/*
+	 * Switch to HSI and disable the PLL before configuration.
+	 *
```
Comment (galak) [inline drivers/clock_control/stm32_ll_clock.c]: can you add why we are switching to HSI before disabling the PLL in the comment

**Review by galak — CHANGES_REQUESTED** [review]
Minor comment fix, otherwise looks good

**Comment by mbolivar** [comment]
> Minor comment fix, otherwise looks good

Thanks for the reviews; now updated.

**Review by galak — APPROVED** [review]

**Pull request merged**